### PR TITLE
Move network code to a shared SampleGraphRequest class

### PR DESCRIPTION
**Pull request by RPangrle**
This will make it easier to reuse for the task retrieval stuff.

### CODE REVIEW

File: Samples/ios/SampleApp/SampleAppiOS/SampleGraphRequest.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// For these errors the error will be a HTTP error code, and the userInfo dictionary will be the
+// error dictionary from the JSON response in the body (if any)
+extern const NSErrorDomain SampleGraphErrorDomain;
+
+@interface SampleGraphRequest : NSObject
+
++ (instancetype)requestWithToken:(NSString *)token;
+
+- (void)getJSON:(NSString *)path
+          block:(void(^)(NSDictionary *json, NSError *error))completionBlock;
```
Comment (unpluggedk) [inline Samples/ios/SampleApp/SampleAppiOS/SampleGraphRequest.h]: nit: being a sample, can we change this to completionBlock: ?

File: Samples/ios/SampleApp/SampleAppiOS/SampleGraphRequest.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- (void)getData:(NSString *)path
+          block:(void (^)(NSData *, NSError *))completionBlock;
```
Comment (unpluggedk) [inline Samples/ios/SampleApp/SampleAppiOS/SampleGraphRequest.h]: same as above, just to make it a little clearer

File: Samples/ios/SampleApp/SampleAppiOS/SampleGraphRequest.m [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    SampleGraphRequest *req = [self new];
+    req->_token = token;
+    return req;
+}
+
+- (void)getData:(NSString *)path
+          block:(void (^)(NSData *, NSError *))completionBlock
+{
+    NSMutableURLRequest *urlRequest = [NSMutableURLRequest new];
+    urlRequest.URL = [NSURL URLWithString:[NSString stringWithFormat:@"https://graph.microsoft.com/beta/%@", path]];
```
Comment (unpluggedk) [inline Samples/ios/SampleApp/SampleAppiOS/SampleGraphRequest.m]: Can we define the graph url to make it more visible at the top?

Comment (RPangrle) [inline Samples/ios/SampleApp/SampleAppiOS/SampleGraphRequest.m]: It's a format string. Format strings should not be defined anywhere but where they are used.

Comment (RPangrle) [inline Samples/ios/SampleApp/SampleAppiOS/SampleGraphRequest.m]: I'll move it into a helper method

**Pull request merged**